boards/raspberrypi/rpi_pico_2/rpi_pico_2-pinctrl.dtsi

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, Andrew Featherstone
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
```

Comment:
We need to create rpi-pico-rp2350a-pinctrl.h

https://github.com/zephyrproject-rtos/zephyr/pull/77859/commits/04c4c43954cc6456efa5501bf3f02040c380aad0